packages/dev/addons/src/navigation/factory/common.ts

```diff
@@ -0,0 +1,95 @@
+import { _LoadScriptModuleAsync } from "core/Misc/tools.internals";
+import type { Mesh } from "core/Meshes/mesh";
+
+import type { INavMeshParametersV2, RecastInjection } from "../types";
+import { GenerateNavMesh } from "../generator";
+import type { RecastNavigationJSPluginV2 } from "../plugin/RecastNavigationJSPlugin";
+
+export let BjsRecast: RecastInjection;
```

Comment:
Seems like this should be nullable.

Reply:
What about exporting an accessor function instead (like `GetBjsRecast` or something) that is synchronous but throws if the value is not set. Then you'd just have a nullable local module scoped variable (which is correctly typed), and usages of it don't need to do any checks or asynchrony, it will just throw if it is null. It's effectively a runtime assertion to protect against programming errors.

Reply:
Backward compatibility with what though? This is a whole new file and a new module level export. What existing thing are you thinking of when you say "backwards compatibility?"

Reply:
Can you please have a look `recastJSPlugin.ts:14`? I was trying to mimic the same behavior. The global variable is set when the Playground loads. This should be the same with V2. If you still think we should use a getter/setter approach  - I am in :-)

Reply:
Well even the global variable names are different, so calling code would have to change in some way. Since that is the case, I personally think switching to safer get/set functions would make sense.

Reply:
The global variable are not the same yet because V2 and V1 will coexist for now. If V2 will replace V1 the global variables must match. I'll change the code to get/set and as you said we will need to update the calling code as well.